main.go

```diff
@@ -474,8 +486,22 @@ func secureNetwork(testNetwork *network.Network, decelerationFactor float64) {
 }
 
 func startSecurityWorker(peer *network.Peer, pace time.Duration) {
-	for range time.Tick(pace) {
-		sendMessage(peer)
+	if pace == time.Duration(0) {
+		return
+	}
+	ticker := time.NewTicker(pace)
+
+	for {
+		select {
+		case <-ticker.C:
+			if config.IMIF == "poisson" {
+				pace = time.Duration(float64(time.Second) * rand.ExpFloat64() / float64(pace))
+				if pace > 0 {
+					ticker = time.NewTicker(pace)
```

Comment:
- Need to stop the ticker before creating a new ticker, or it will pressure the garbage collector and slow down the simulation
- The more efficient way is to use ticker.Reset(), instead of creating new tickers frequently